src/main.rs

```diff
@@ -62,7 +63,34 @@ async fn main() -> Result<()> {
     let url = Url::new(args.url, args.default_scheme)?;
     let host = url.host().ok_or_else(|| anyhow!("Missing hostname"))?;
     let method = args.method.unwrap_or_else(|| Method::from(&body)).into();
-    let auth = Auth::new(args.auth, args.auth_type, &host)?;
+    let mut auth = Auth::new(args.auth, args.auth_type, &host)?;
+
+    // load previous session if present
+    let previous_session = match args.session {
+        None => None,
+        Some(ref identifier) => {
+            match Session::load(identifier, &host) {
+                Err(why) => panic!("couldn't load session {}: {}", &identifier, why),
+                Ok(result) => result,
+            }
+        },
+    };
+    let session_for_merge = previous_session.clone();
+    // Use auth from previous session if no auth present
+    if let (true, Some(p)) = (auth.is_none(), previous_session) {
+        auth = p.auth;
+    }
+    let saved_auth = auth.clone();
+    // Merge headers from parameters and previous session
+    let (headers, headers_to_unset) = request_items.headers(session_for_merge.as_ref())?;
```

Comment:
I would have liked it if we could limit `request_items.rs`'s responsibility to parsing current request_items.
1. `request_items.headers()` would produce `reqwest::header::HeaderMap` from user input.
2. `session.rs` would construct `reqwest::header::HeaderMap` from prev session.
3. `session.rs` would take the two headers and merge them.